api/api.go

```diff
@@ -986,28 +986,47 @@ func (c *Client) websocket(endpoint string, q *QueryOptions) (*websocket.Conn, *
 	conn, resp, err := dialer.Dial(rhttp.URL.String(), rhttp.Header)
 
 	// check resp status code, as it's more informative than handshake error we get from ws library
-	if resp != nil && resp.StatusCode != http.StatusSwitchingProtocols {
-		var buf bytes.Buffer
+	if resp != nil {
+		switch resp.StatusCode {
+		case http.StatusSwitchingProtocols:
+			// Connection upgrade was successful.
+
+		case http.StatusPermanentRedirect, http.StatusTemporaryRedirect, http.StatusMovedPermanently:
+			locs := resp.Header["Location"]
+			if len(locs) == 0 {
+				return nil, nil, errors.New("redirect request missing Location header")
```

Comment:
Do we want to quote the `Location` entry to make it more obvious that's the header we are referring to?